### PR TITLE
Show LoadingIndicator while waiting for the realm to switch in AccountPickScreen

### PR DESCRIPTION
**Pull request by agzuniverse**
Navigate to loading screen to fill during the delay between tapping on a realm and switching to it.

### CODE REVIEW

**Comment by borisyankov** [comment]
This looks a bit weird from UI standpoint. The progress indicator is disconnected from the item clicked.
One option would be to turn the item itself into a spinner... but there is a simpler solution:
Navigate to the `loading` route which is what you see on startup.

A separate thing would be to investigate why it takes so long on switching. Maybe we can speed it up?

**Comment by borisyankov** [comment]
About my suggestion: dispatching a `navigateToLoading` action should be enough to achieve it.

**Comment by agzuniverse** [comment]
Made that change, that indeed seems better.

I'll see if I can find any specific reason that affects the performance but I'd assume its due to slow network and/or performance issues of react native on low end devices.

File: src/account/AccountPickScreen.js [hunk 1]
```diff
@@ -27,6 +32,7 @@ class AccountPickScreen extends PureComponent<Props> {
 
   handleAccountSelect = (index: number) => {
     const { accounts, dispatch } = this.props;
+    dispatch(navigateToLoading());
```
Comment (jainkuniya) [inline src/account/AccountPickScreen.js]: Now there are two pushes of `LoadingScreen` route.

**Comment by borisyankov** [comment]
It is worth exploring why there is a delay on switching accounts.

@agzuniverse if you are wondering why there is another switch to the 'loading' screen have a look at this:
https://github.com/zulip/zulip-mobile/blob/master/src/nav/navReducers.js#L66

I think it is quite useful as a learning experience to debug/profile why it takes so long to switch accounts. (@jainkuniya can help you but would be good to examine this yourself first)

**Comment by agzuniverse** [comment]
To remove the original call, setting case `ACCOUNT_SWITCH`  to `initialState` causes indefinite loading screen. Setting it to `NULL_NAV_STATE` causes an error.

**Comment by borisyankov** [comment]
It looks like we have implemented this already and it works :)
This PR can now be closed @gnprice

**Pull request closed without merging**